### PR TITLE
`rustc_ast_{passes,pretty}`: remove `ref` patterns

### PR DESCRIPTION
**Pull request by WaffleLapkin**
r? @compiler-errors 
Previous PR: https://github.com/rust-lang/rust/pull/104721

### CODE REVIEW

**Review by jyn514 — APPROVED** [review]

**Comment by jyn514** [comment]
@bors delegate=WaffleLapkin

(Not sure if you wanted a review from Michael too)

File: compiler/rustc_ast_pretty/src/pprust/state.rs [hunk 1]
```diff
@@ -64,6 +64,7 @@ impl<'a> Comments<'a> {
         Comments { sm, comments, current: 0 }
     }
 
+    // This shouldn't probably clone lmao
```
Comment (compiler-errors) [inline compiler/rustc_ast_pretty/src/pprust/state.rs:67]:
```suggestion
    // FIXME: This shouldn't probably clone lmao
```

**Comment by compiler-errors** [comment]
r=me after committing nit (either via the button on github or amending ur commit, i dont care)

**Pull request merged**